src/main/java/io/jenkins/plugins/synopsys/security/scan/service/ScannerArgumentService.java

```diff
@@ -253,6 +273,56 @@ private void setScanObject(
             bridgeInput.setCoverity(coverity);
         } else if (scanObject instanceof Polaris) {
             handlePolarisScan(bridgeInput, (Polaris) scanObject, scmObject, sarifObject, scanParameters);
+        } else if (scanObject instanceof SRM) {
+            SRM srm = (SRM) scanObject;
+            if (scmObject != null) {
+                if (Utility.isStringNullOrBlank(srm.getProject().getName())
+                        && Utility.isStringNullOrBlank(srm.getProject().getId())) {
+                    setDefaultValueSrmParams(srm, scmObject);
+                }
+            }
+            BlackDuck blackDuck = handleScaArbitary(bridgeInput, scanParameters);
+            Coverity coverity = handleSastArbitary(bridgeInput, scanParameters);
+            handleSrmSCAInstallationPath(bridgeInput, scanParameters, blackDuck);
+            handleSrmSASTInstallationPath(bridgeInput, scanParameters, coverity);
+            bridgeInput.setSrm(srm);
+        }
+    }
+
+    private void handleSrmSCAInstallationPath(
+            BridgeInput bridgeInput, Map<String, Object> scanParameters, BlackDuck blackDuck) {
+        if (scanParameters.containsKey(ApplicationConstants.SRM_SCA_EXECUTION_PATH_KEY)) {
+            String installationPath = scanParameters
+                    .get(ApplicationConstants.SRM_SCA_EXECUTION_PATH_KEY)
+                    .toString()
+                    .trim();
+            if (installationPath != null && !installationPath.isBlank()) {
+                if (blackDuck == null) blackDuck = new BlackDuck();
+                io.jenkins.plugins.synopsys.security.scan.input.blackduck.Execution execution =
+                        new io.jenkins.plugins.synopsys.security.scan.input.blackduck.Execution();
+                execution.setPath(installationPath);
+                blackDuck.setExecution(execution);
+                bridgeInput.setBlackDuck(blackDuck);
+            }
+        }
+    }
+
+    private void handleSrmSASTInstallationPath(
+            BridgeInput bridgeInput, Map<String, Object> scanParameters, Coverity coverity) {
+
+        if (scanParameters.containsKey(ApplicationConstants.SRM_SAST_EXECUTION_PATH_KEY)) {
+            String installationPath = scanParameters
+                    .get(ApplicationConstants.SRM_SAST_EXECUTION_PATH_KEY)
+                    .toString()
+                    .trim();
+            if (installationPath != null && !installationPath.isBlank()) {
+                if (coverity == null) coverity = new Coverity();
+                io.jenkins.plugins.synopsys.security.scan.input.coverity.Execution execution =
+                        new io.jenkins.plugins.synopsys.security.scan.input.coverity.Execution();
+                execution.setPath(installationPath);
```

Comment:
Same here!

Reply:
There are two Class with same "Execution" name one for blackduck and one for coverity